mm2src/coins/eth/v2_activation.rs

```diff
@@ -488,6 +491,30 @@ impl EthCoin {
 
         let conf = coin_conf(&ctx, &ticker);
 
+        let web3_instances: Vec<Web3Instance> = self
+            .web3_instances
+            .lock()
+            .await
+            .iter()
+            .map(|node| {
+                let mut transport = node.web3.transport().clone();
+                if let Some(auth) = transport.proxy_auth_validation_generator_as_mut() {
+                    auth.coin_ticker = ticker.clone();
+                }
+                let web3 = Web3::new(transport);
+                Web3Instance {
+                    web3,
+                    is_parity: node.is_parity,
+                }
+            })
+            .collect();
+
+        let required_confirmations = AtomicU64::new(
+            conf["required_confirmations"]
+                .as_u64()
+                .unwrap_or_else(|| self.required_confirmations.load(Ordering::Relaxed)),
+        );
+
```

Comment:
sorry, I've got a quick question here 

What if you clone minimal necessary data while holding the lock?

```rust
    let cloned_data: Vec<_> = self.web3_instances.lock().await
        .iter()
        .map(|node| (node.web3.transport().clone(), node.is_parity))
        .collect();
```

Reply:
we are locking web instances to  collect new vector of `Web3Instance` with correct coin ticker in validation generator
```rust
pub struct ProxyAuthValidationGenerator {
    pub coin_ticker: String,
    pub secret: Secret,
    pub address: String,
}
```
this structure will be used to generate signed message for proxy layer, 
https://github.com/KomodoPlatform/komodo-defi-proxy/blob/main/src/security/sign.rs#L30
where ticker will be used in rate_limiter_key
https://github.com/KomodoPlatform/komodo-defi-proxy/blob/main/src/proxy/quicknode.rs#L187
also in logs
https://github.com/KomodoPlatform/komodo-defi-proxy/blob/main/src/proxy/quicknode.rs#L211-L237

Reply:
thanks!